src/core/dynamic_batch_scheduler.cc

```diff
@@ -582,12 +582,24 @@ DynamicBatchScheduler::DelegateResponse(
   completion_queue_.emplace_back();
   auto queue_slot = &completion_queue_.back();
   uint64_t request_hash = request->CacheKey();
+  // Pass raw ptr to lambda for tracking stats from cache and updating
+  // metric reporter on cache miss stats after insertion
+  InferenceRequest* raw_request_ptr = request.get();
+
   request->SetResponseDelegator(
-      [this, queue_slot, request_hash](
+      [this, queue_slot, request_hash, raw_request_ptr](
```

Comment:
Not sure if there's a better way to do this. Passing the unique_ptr has an ownership/deletion issue